sunpy/map/sources/goes.py

```diff
@@ -0,0 +1,92 @@
+    range near 100. A set of metallic thin-film filters provides temperature
+    discrimination in the 0.6–6.0 nm bandpass. The spatial resolution of
+    approximately 10 arcsec Full width half maximum (FWHM) is sampled with 5
+    arcsec pixels.
+
+    SXI was first launched on 23 July 2001 on NOAA’s GOES-12 satellite and has
+    obtained uninterrupted images starting on 22 January 2003.
+
+    References
+    ----------
+    * `Mission Instrument Page <https://sxi.ngdc.noaa.gov/index.html>`_
+    * `Data notes and Fits headers <https://sxi.ngdc.noaa.gov/sxi_data_notes.html>`_
+    * `Official SXI Data Book <https://sxi.ngdc.noaa.gov/images/section06.pdf>`_
+    * `Instrument Paper DOI: 10.1007/s11207-005-7416-x <https://link.springer.com/article/10.1007/s11207-005-7416-x>`_
+    * `SXI Performance Paper DOI: 10.1007/s11207-005-7417-9 <https://link.springer.com/article/10.1007/s11207-005-7417-9>`_
+    """
+
+    filter_wheel_measurements = ["OPEN", "PTHNA", "PMEDA", "PTHK",
+                                 "BTHNA", "BMED", "BTHK", "UV", "RDSH"]
+
+    def __init__(self, data, header, **kwargs):
+
+        # check if the filter setting is in the list of expected values
+        fw = header.get('WAVELNTH')
+        if fw.lower().strip() not in [item.lower() for item in self.filter_wheel_measurements]:
+            raise ValueError('Unexpected filter setting in header.')
+
+        # the following are necessary fixes to the header
+        header['DETECTOR'] = 'SXI'
+        # save the filter setting
+        header['FILTER'] = header.get('WAVELNTH')
+        # update the wavelength with a number since a str is not supported.
+        # TODO: change the number according to the filter
+        header['WAVELNTH'] = 14
+        header['WAVEUNIT'] = 'Angstrom'
+        # need to switch these as they are not correct - Hannah
+        crota2 = header['CROTA1']
+        crota1 = header['CROTA2']
+        header['CROTA1'] = crota1
+        header['CROTA2'] = crota2
+        header['CTYPE1'] = "HPLN-TAN"
+        header['CTYPE2'] = "HPLT-TAN"
+        obs_time = header['date_obs']
+        header['HGLT_OBS'] = get_sun_B0(obs_time).to('deg').value
+        header['HGLN_OBS'] = 0
+        header['RSUN_OBS'] = sun.solar_semidiameter_angular_size(obs_time).to('arcsec').value
+        header['RSUN_REF'] = sun.constants.radius.to('m').value
+        header['DSUN_OBS'] = get_sunearth_distance(obs_time).to('m').value
+
+        GenericMap.__init__(self, data, header, **kwargs)
+
+        self._nickname = "{0}/{1}".format(self.meta['telescop'], self.detector)
+        # Colour maps
+        self.plot_settings['cmap'] = cm.get_cmap('sohoeit195')
+        self.plot_settings['norm'] = LogNorm(vmin=1, vmax=1e4)
+
+    @classmethod
+    def is_datasource_for(cls, data, header, **kwargs):
+        """Determines if header corresponds to an GOES SXI image"""
+        return header.get('instrume').count('SXI')
```

Comment:
I still don't understand why you would use `count`. I would expect the output of `is_datasource_for()` a `bool` and that produces a number. At least you could return the `bool()` of that

Reply:
Ah! Sorry I understand your issue here. The reason I did not want to check for the value itself is because I currently see it as "SXI-FM1". I'm not sure what the FM1 refers to and I worry that it may change for GOES-12 or GOES-13. I'll change this to return a bool.

Reply:
The way I suggested yesterday:
```python
return "SXI" in header.get('instrume', 'who knows')
```
will check whether `SXI` is in the string, not for the whole value. The `'who knows'` bit will be in case the keyword `'instrume'` doesn't exist and you get `None` instead. By putting `'who knows`' it's clear that we are trying to guess the instrument name.
Of course, maybe `'unkown instrument`' is a better wording than `'who knows'`.

Reply:
or just an empty string would also work :wink:

Reply:
but an empty string is more cryptic :stuck_out_tongue_closed_eyes:

Reply:
hardly `"SXI"` is obviously not in `""`